File: 4. Maths-I Big Integers/julka_spoj.py
t=10
while t!=0:
    total = int(input())
    excess = int(input())

    # Klaudia + Natalia = total
    # Klaudia = Natalia + excess
    # So, Natalia + excess + Natalia = total
    # Natalia = (total-excess)/2

    natalia = (total-excess)//2
    klaudia = natalia + excess

    print(klaudia)
    print(natalia)
    t-=1

File: 04. Maths-I Big Integers/julka_spoj.java
import java.util.*;
import java.lang.*;
import java.math.BigInteger;

class julka_spoj
{
	public static void main (String[] args) throws java.lang.Exception
	{
		Scanner s = new Scanner(System.in);

        int t = 10;
        for(int i=0; i<10; i++){
            BigInteger total = s.nextBigInteger();
            BigInteger excess = s.nextBigInteger();

            apples_count(total,excess);
        }
	}
	
	public static void apples_count(BigInteger total, BigInteger excess){


        BigInteger natalia = (total.subtract(excess)).divide(BigInteger.valueOf(2));
        BigInteger klaudia = natalia.add(excess);

        System.out.println(klaudia);
        System.out.println(natalia);
    }
}